transcom/mymove
# Add Patch handler to Public Shipment handler

## Description

This PR adds a patch handler to the public shipments handler to allow the TSP user to update a shipment with pre move survey values. If the TSP user is not authorized to view the shipment, the shipment is not updated. Does not take into account office users.

## Setup

1. 
```
make db_dev_reset 
make db_dev_migrate
bin/generate-test-data -scenario=7
```
-> to add fields to shipment table and generate shipments with associated shipment offers with "awarded" statuses.

2. Go to "tsplocal:3000", verify shipments appear in the queue.
3. Go to "tsplocal:3000/api/v1/docs". Patch a shipment by id (find shipment id from shipment in queue list in previous step). You should only pass in values prefixed by "premove", as other values are not handled.

Verify 200 response.

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159783566) for this change
